### ISSUE TITLE
500 Error for unknown reason

### ISSUE THREAD

**Issue by stone12379** [IIIF/image-validator#87]
https://iiif.io/api/image/validator/service/region_pixels?version=2.0&server=common.image2.fiii.cc&prefix=&identifier=f72056b8559f390901112923d48958d6
@glenrobson

**Comment by glenrobson** [comment]
I've fixed the 500 error but I don't know what is causing the test to fail... It looks like Pillow can't open the following images:

http://common.image2.fiii.cc/f72056b8559f390901112923d48958d6/full/,729/0/default.jpg
http://common.image2.fiii.cc/f72056b8559f390901112923d48958d6/500,800,100,100/58,58/0/default.jpg

with the following error:

```
cannot identify image file <_io.BytesIO object at 0x1021c84c0>
```

With this line of code:
https://github.com/IIIF/image-validator/blob/bf7026592ba0576c91e31f97389f9e1c33dcfd28/iiif_validator/validator.py#L328

but I can't see why it won't open this image. Using curl and identify the image looks ok...

**Comment by stone12379** [comment]
> I've fixed the 500 error but I don't know what is causing the test to fail... It looks like Pillow can't open the following images:
> 
> http://common.image2.fiii.cc/f72056b8559f390901112923d48958d6/full/,729/0/default.jpg
> http://common.image2.fiii.cc/f72056b8559f390901112923d48958d6/500,800,100,100/58,58/0/default.jpg
> 
> with the following error:
> 
> ```
> cannot identify image file <_io.BytesIO object at 0x1021c84c0>
> ```
> 
> With this line of code:
> https://github.com/IIIF/image-validator/blob/bf7026592ba0576c91e31f97389f9e1c33dcfd28/iiif_validator/validator.py#L328
> 
> but I can't see why it won't open this image. Using curl and identify the image looks ok...

Thank you Glen for the quick response. It seems to be a known issue in pillow. See https://stackoverflow.com/questions/19230991/image-open-cannot-identify-image-file-python

**Comment by stone12379** [comment]
Glen, I figured out the root cause. Http response with body encoded in base64(see below) would cause Pillow to fail. I've managed to change the way to return from server side. But if you want to fix in the validator, try the solution above or try this one: [](https://stackoverflow.com/questions/31077366/pil-cannot-identify-image-file-for-io-bytesio-object). Hopefully it could be resolved. 

    response.statusCode = 200
    response.headers['Content-Type'] = mime.lookup(format)

    response.isBase64Encoded = true
    response.body = data.toString('base64')
